<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ConnectDialog</class>
 <widget class="QDialog" name="ConnectDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>692</width>
    <height>552</height>
   </rect>
  </property>
  <property name="mouseTracking">
   <bool>true</bool>
  </property>
  <property name="focusPolicy">
   <enum>Qt::TabFocus</enum>
  </property>
  <property name="windowTitle">
   <string>Egge's Messenger: Connect</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QTabWidget" name="tabConnect">
     <property name="maximumSize">
      <size>
       <width>601</width>
       <height>461</height>
      </size>
     </property>
     <property name="currentIndex">
      <number>2</number>
     </property>
     <property name="usesScrollButtons">
      <bool>false</bool>
     </property>
     <property name="documentMode">
      <bool>false</bool>
     </property>
     <property name="tabsClosable">
      <bool>false</bool>
     </property>
     <property name="movable">
      <bool>true</bool>
     </property>
     <property name="tabBarAutoHide">
      <bool>false</bool>
     </property>
     <widget class="QWidget" name="tabClient">
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="cursor">
       <cursorShape>PointingHandCursor</cursorShape>
      </property>
      <property name="mouseTracking">
       <bool>true</bool>
      </property>
      <attribute name="title">
       <string>Client</string>
      </attribute>
      <widget class="QLabel" name="labelClientInfo">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>20</y>
         <width>561</width>
         <height>17</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">QLabel {font-style: italic}</string>
       </property>
       <property name="text">
        <string>Create connection to server as client</string>
       </property>
      </widget>
      <widget class="QLabel" name="labelServerIP">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>60</y>
         <width>221</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Server IP number:</string>
       </property>
      </widget>
      <widget class="QLabel" name="labelServerPortNumber">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>100</y>
         <width>221</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Server port number:</string>
       </property>
      </widget>
      <widget class="QComboBox" name="comboBoxServerIP">
       <property name="geometry">
        <rect>
         <x>260</x>
         <y>60</y>
         <width>301</width>
         <height>25</height>
        </rect>
       </property>
       <property name="cursor">
        <cursorShape>PointingHandCursor</cursorShape>
       </property>
       <property name="mouseTracking">
        <bool>true</bool>
       </property>
       <property name="editable">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QComboBox" name="comboBoxServerPortNumber">
       <property name="geometry">
        <rect>
         <x>260</x>
         <y>100</y>
         <width>301</width>
         <height>25</height>
        </rect>
       </property>
       <property name="cursor">
        <cursorShape>PointingHandCursor</cursorShape>
       </property>
       <property name="editable">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButtonConnect2Server">
       <property name="geometry">
        <rect>
         <x>440</x>
         <y>200</y>
         <width>121</width>
         <height>40</height>
        </rect>
       </property>
       <property name="text">
        <string>Connect</string>
       </property>
      </widget>
      <widget class="QLabel" name="labelClientInfo_2">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>160</y>
         <width>541</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>(Currently no connection.)</string>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="tabServer">
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <attribute name="title">
       <string>Server</string>
      </attribute>
      <widget class="QListWidget" name="listWidgetHostInfo">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>60</y>
         <width>241</width>
         <height>201</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>8</pointsize>
        </font>
       </property>
      </widget>
      <widget class="QGroupBox" name="groupBoxServerOnOff">
       <property name="geometry">
        <rect>
         <x>280</x>
         <y>60</y>
         <width>301</width>
         <height>121</height>
        </rect>
       </property>
       <property name="title">
        <string>Start server on this host?</string>
       </property>
       <widget class="QRadioButton" name="radioButtonStartServerOK">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>40</y>
          <width>241</width>
          <height>23</height>
         </rect>
        </property>
        <property name="toolTip">
         <string>Accept providing server capability on this host. </string>
        </property>
        <property name="text">
         <string>Yes, I'm cool with it!</string>
        </property>
       </widget>
       <widget class="QRadioButton" name="radioButtonStartServerNAK">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>60</y>
          <width>261</width>
          <height>23</height>
         </rect>
        </property>
        <property name="toolTip">
         <string>No server capability will be provided on this host.</string>
        </property>
        <property name="text">
         <string>Sorry, I'm otherwise ... ahem ... engaged.</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </widget>
      <widget class="QPushButton" name="pushButtonLaunchServer">
       <property name="geometry">
        <rect>
         <x>460</x>
         <y>360</y>
         <width>121</width>
         <height>40</height>
        </rect>
       </property>
       <property name="text">
        <string>Launch</string>
       </property>
      </widget>
      <widget class="QLabel" name="labelServerInfo">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>20</y>
         <width>381</width>
         <height>17</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">QLabel {font-style: italic}</string>
       </property>
       <property name="text">
        <string>Provide server functionality from this host</string>
       </property>
      </widget>
      <widget class="QFrame" name="frame">
       <property name="geometry">
        <rect>
         <x>280</x>
         <y>200</y>
         <width>301</width>
         <height>61</height>
        </rect>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <widget class="QLabel" name="labelServerPortNumber_2">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>20</y>
          <width>221</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>Server port number:</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="lineEditServerPortNumber">
        <property name="geometry">
         <rect>
          <x>160</x>
          <y>20</y>
          <width>113</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
      </widget>
      <widget class="QLabel" name="labelServerInfo_2">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>280</y>
         <width>561</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="tabTorControl">
      <attribute name="title">
       <string> Tor Control </string>
      </attribute>
      <widget class="QLabel" name="labelTorControlInfo">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>17</y>
         <width>561</width>
         <height>40</height>
        </rect>
       </property>
       <property name="text">
        <string>Tor control and oversight</string>
       </property>
      </widget>
      <widget class="QListWidget" name="listWidgetTorControlHistory">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>60</y>
         <width>561</width>
         <height>261</height>
        </rect>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="tabTorClient">
      <property name="focusPolicy">
       <enum>Qt::TabFocus</enum>
      </property>
      <attribute name="title">
       <string> Tor Client </string>
      </attribute>
      <widget class="QComboBox" name="comboBoxTorClientServer">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>100</y>
         <width>401</width>
         <height>25</height>
        </rect>
       </property>
       <property name="cursor">
        <cursorShape>OpenHandCursor</cursorShape>
       </property>
       <property name="editable">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QLabel" name="labelTorClientInfo">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>20</y>
         <width>541</width>
         <height>21</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <italic>true</italic>
        </font>
       </property>
       <property name="text">
        <string>Connect to server via Tor</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButtonTorClientConnect">
       <property name="geometry">
        <rect>
         <x>440</x>
         <y>100</y>
         <width>80</width>
         <height>25</height>
        </rect>
       </property>
       <property name="cursor">
        <cursorShape>PointingHandCursor</cursorShape>
       </property>
       <property name="mouseTracking">
        <bool>true</bool>
       </property>
       <property name="text">
        <string>Connect</string>
       </property>
      </widget>
      <widget class="QListWidget" name="listWidgetTorClientFeedback">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>140</y>
         <width>501</width>
         <height>261</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="labelTorClientInfo2">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>60</y>
         <width>541</width>
         <height>17</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <italic>true</italic>
        </font>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </widget>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QDialogButtonBox" name="buttonBoxConnectOkAbort">
     <property name="standardButtons">
      <set>QDialogButtonBox::Abort|QDialogButtonBox::Ok</set>
     </property>
     <property name="centerButtons">
      <bool>false</bool>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
